civicrm/custom/php/CRM/NYSS/BAO/Integration/WebsiteEvent/SurveyEvent.php

```diff
@@ -44,19 +44,46 @@ public function process(int $contact_id): static {
     return $this;
   }
 
+  /**
+   * @param int $contact_id
+   *
+   * @return void
+   * @throws \CRM_Core_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
   protected function processForm(int $contact_id): void {
-    // form_id, form_values, form_title, detail??
-    $params = new stdClass();
-    $params->form_id = $this->getFormId();
-    $params->form_values = $this->getFormFields();
-    $params->form_title = $this->getFormTitle();
-    $params->detail = $this->getEventDetails();
-    $params->created_at = $this->getEventData()->getCreatedAtAsDateTime();
-
-    $result = CRM_NYSS_BAO_Integration_Website::processSurvey($contact_id, $this->getEventAction(), $params);
-    if ($result['is_error'] == 1) {
-      throw new Exception("Error processing webform/survey: " . $result['error_message']);
-    }
+
+    // NYSS 16799 -- No more custom field groups for surveys. Now, we just add
+    // to Activity.details and Website_Survey.survey_data
+    // $result = CRM_NYSS_BAO_Integration_Website::processSurvey($contact_id, $this->getEventAction(), $params);
+
+    // create activity
+    $activity_type_id = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'Website Survey');
+    $event_info = $this->getEventInfo();
+    $form_fields = $event_info->form_values;
+    $fields_as_text = array_reduce($form_fields, function ($carry, $field) {
+      return $carry . self::formatFieldsAsText($field->field, $field->value);
+    });
+
+    $results = \Civi\Api4\Activity::create($this->getCiviPermissionCheck())
+      ->addValue('activity_type_id', $activity_type_id)
```

Comment:
you should be able to reference activity_type_id:name rather than retrieve the ID.